boa_engine/src/object/jsobject.rs

```diff
@@ -33,32 +33,50 @@ pub struct JsObject {
 }
 
 impl JsObject {
-    /// Create a new `JsObject` from an internal `Object`.
-    fn from_object(object: Object) -> Self {
-        Self {
-            inner: Gc::new(GcCell::new(object)),
-        }
+    /// Creates a new ordinary object with its prototype set to the `Object` prototype.
+    ///
+    /// In spec jargon, this is equivalent to calling the
+    /// [`OrdinaryObjectCreate(%Object.prototype%)`][call] operation.
+    ///
+    /// [call]: https://tc39.es/ecma262/#sec-ordinaryobjectcreate
+    #[inline]
+    #[must_use]
+    pub fn new(context: &mut Context) -> Self {
+        Self::from_proto_and_data(
+            context.intrinsics().constructors().object().prototype(),
+            ObjectData::ordinary(),
+        )
     }
 
-    /// Create a new empty `JsObject`, with `prototype` set to `JsValue::Null`
-    /// and `data` set to `ObjectData::ordinary`
+    /// Creates a new ordinary object, with its prototype set to null.
+    ///
+    /// In spec jargon, this is equivalent to calling the [`OrdinaryObjectCreate(null)`][call]
+    /// operation.
+    ///
+    /// [call]: https://tc39.es/ecma262/#sec-ordinaryobjectcreate
+    #[inline]
     #[must_use]
-    pub fn empty() -> Self {
-        Self::from_object(Object::default())
+    pub fn with_null_proto() -> Self {
```

Comment:
Small question here: should the method here be changed to something like `JsObject::create_null` (this is also  `JsObject::new`)? 

The way I'm thinking about this is that there are two operations that are the equivalent of `OrdinaryObjectCreate(%prototype%)`, but the methods' names don't currently link them as similar operations for `JsObject`.

Reply:
Maybe `JsObject::new` -> `JsObject::with_object_proto`, `JsObject::with_null_proto` stays the same, and `JsObject::from_proto_and_data` -> `JsObject::with_proto_and_data`? That way we follow the "standard" Rust naming while also linking the names to the common operation `OrdinaryObjectCreate`.

Reply:
The first two sound good to me. Not sure about the last change.

I actually think `JsObject::from_proto_and_data` makes sense, primarily because the `with_` clauses are mainly just creating an ordinary object data with the designated prototype. I believe `from_proto_and_data` is used heavily internally in connection with `OrdinaryObjectCrate`, but it also serves as a good API method for a `JsObject` to be created with the provided prototype and data.

Reply:
Agreed. `with_` functions which are just utility wrappers around `from_proto_and_data` sounds better.